modin/experimental/xgboost/xgboost_ray.py

```diff
@@ -420,129 +344,106 @@ def _assign_row_partitions_to_actors(
-                            num_actors_with_max_parts -= 1
-                        else:
-                            pop_slice = slice(0, min_parts_per_actor)
-
-                        row_partitions_by_actors[rank].extend(
-                            init_parts_distribution[actor_ip][pop_slice]
-                        )
-                        # Delete parts which we already assign
-                        del init_parts_distribution[actor_ip][pop_slice]
-                    else:
-                        row_partitions_by_actors[rank].extend(
-                            init_parts_distribution[actor_ip]
-                        )
-                        init_parts_distribution[actor_ip] = []
-
-            # Remove empty IPs
-            for ip in list(init_parts_distribution):
-                if len(init_parts_distribution[ip]) == 0:
-                    init_parts_distribution.pop(ip)
-
-            # IP's aren't necessary now
-            init_parts_distribution = [
-                pair for pairs in init_parts_distribution.values() for pair in pairs
-            ]
-
-            # Fill the actors with extra parts (movements data between nodes)
-            for rank in range(len(actors)):
-                num_parts_on_rank = len(row_partitions_by_actors[rank])
-
-                if num_parts_on_rank == max_parts_per_actor or (
-                    num_parts_on_rank == min_parts_per_actor
-                    and num_actors_with_max_parts == 0
-                ):
-                    continue
-
-                if num_actors_with_max_parts > 0:
-                    pop_slice = slice(0, max_parts_per_actor - num_parts_on_rank)
-                    num_actors_with_max_parts -= 1
-                else:
-                    pop_slice = slice(0, min_parts_per_actor - num_parts_on_rank)
-
-                row_partitions_by_actors[rank].extend(
-                    init_parts_distribution[pop_slice]
-                )
-                del init_parts_distribution[pop_slice]
-
-            if len(init_parts_distribution) != 0:
-                raise RuntimeError(
-                    f"Not all partitions were ditributed between actors: {len(init_parts_distribution)} left."
-                )
-
-            row_parts_by_ranks = dict()
-            for rank, pairs_part_pos in dict(row_partitions_by_actors).items():
-                parts, order = zip(*pairs_part_pos)
-                row_parts_by_ranks[rank] = (list(parts), list(order))
-        else:
-            row_parts_by_ranks = {rank: ([], []) for rank in range(len(actors))}
-
-            for rank, (_, order_of_indexes) in data_for_aligning.items():
-                row_parts_by_ranks[rank][1].extend(order_of_indexes)
-                for row_idx in order_of_indexes:
-                    row_parts_by_ranks[rank][0].append(row_partitions[row_idx])
-    else:
-        row_parts_by_ranks = defaultdict(list)
-        _, parts_ref = zip(*row_partitions)
+    if data_for_aligning is None:
```

Comment:
This looks like a huge diff. Is the only change relates to removing `if not is_predict:`?

Reply:
The reason of so huge diff is in moving block of code to the left after removing `if no_is_predict` and corresponded `else` branch